### PR TITLE
[Python API] Add_extension test

### PR DESCRIPTION
**Pull request by akuporos**
### Tickets:
 - #60828

### CODE REVIEW

File: inference-engine/ie_bridges/python/tests/test_IECore.py [hunk 1]
```diff
@@ -5,8 +5,9 @@
 import pytest
 from sys import platform
 from pathlib import Path
+import numpy as np
 
-from openvino.inference_engine import IENetwork, IECore, ExecutableNetwork
+from openvino.inference_engine import IENetwork, IECore, ExecutableNetwork, get_library_path
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/python/tests/test_IECore.py:10]: do we need `get_library_path` in public API? I suppose IE core finds extensions located in the same folder by default, is not it?

Comment (akuporos) [reply inference-engine/ie_bridges/python/tests/test_IECore.py:10]: I checked, looks like it's really redundant

File: inference-engine/ie_bridges/python/tests/test_IECore.py [hunk 2]
```diff
@@ -253,3 +254,69 @@ def test_net_from_buffer_valid():
...
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="1" to-port="1"/>
+        <edge from-layer="1" from-port="2" to-layer="2" to-port="0"/>
+    </edges>
+</net>""")
+
+    ie = IECore()
+    ie.add_extension(extension_path="libtemplate_extension.so",
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/python/tests/test_IECore.py:310]: will it work on WIndows?

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx [hunk 3]
```diff
@@ -322,13 +322,15 @@ cdef class IECore:
...
     #  ```
     cpdef IENetwork read_network(self, model: [str, bytes, os.PathLike], weights: [str, bytes, os.PathLike] = "", init_from_buffer: bool = False):
-        cdef uint8_t*bin_buffer
         cdef string weights_
         cdef string model_
         cdef IENetwork net = IENetwork()
         if init_from_buffer:
             model_ = bytes(model)
-            net.impl = self.impl.readNetwork(model_, weights, len(weights))
+            if not weights:
```
Comment (AlexeyLebedev1) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx:330]: Could you please add a test for this case where we call "read_network(model=model_xml, init_from_buffer=True) ?

Comment (akuporos) [reply inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx:330]: I added assert in my test

File: inference-engine/ie_bridges/python/tests/test_IECore.py (hunk 2 above)
Comment (akuporos) [reply inference-engine/ie_bridges/python/tests/test_IECore.py]: good catch, added for win

File: inference-engine/ie_bridges/python/tests/test_IECore.py [hunk 4]
```diff
@@ -253,3 +254,75 @@ def test_net_from_buffer_valid():
...
+    </edges>
+</net>""")
+
+    ie = IECore()
+    if platform == "linux" or platform == "linux2" or platform == "darwin":
+        ie.add_extension(extension_path="libtemplate_extension.so",
+                         device_name="CPU")
+    elif platform == "win32":
+        ie.add_extension(extension_path="template_extension.dll",
+                         device_name="CPU")
```
Comment (ilyachur) [inline inference-engine/ie_bridges/python/tests/test_IECore.py:315]: Looks easier:
```suggestion
    if platform == "win32":
        ie.add_extension(extension_path="template_extension.dll",
                         device_name="CPU")
    else:
        ie.add_extension(extension_path="libtemplate_extension.so",
                         device_name="CPU")
```

File: inference-engine/ie_bridges/python/tests/test_IECore.py [hunk 5]
```diff
@@ -253,3 +254,75 @@ def test_net_from_buffer_valid():
...
+    </edges>
+</net>""")
+
+    ie = IECore()
+    if platform == "win32":
+        ie.add_extension(extension_path="template_extension.dll",
+                         device_name="CPU")
+    else:
+        ie.add_extension(extension_path="libtemplate_extension.so",
+                         device_name="CPU")
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/python/tests/test_IECore.py:315]: device name should be omitted

Comment (akuporos) [reply inference-engine/ie_bridges/python/tests/test_IECore.py]: done

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@akuporos please, make CI green

**Review by ilyachur — APPROVED** [review]

**Review by AlexeyLebedev1 — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
can we finish with this?

**Comment by ilya-lavrenov** [comment]
can we finalize?

**Comment by ilya-lavrenov** [comment]
can we finish with this?

**Comment by ilya-lavrenov** [comment]
@akuporos is it still valid?

**Comment by ilyachur** [comment]
Looks like it is legacy API, I think we can close it.

**Pull request closed without merging**